test_rclcpp/test/test_subscription.cpp

```diff
@@ -140,3 +140,50 @@ TEST(CLASSNAME(test_subscription, RMW_IMPLEMENTATION), subscription_and_spinning
   executor.spin_node_some(node);
   ASSERT_EQ(5, counter);
 }
+
+// Shortened version of the test for the ConstSharedPtr callback signature
+TEST(CLASSNAME(test_subscription, RMW_IMPLEMENTATION), subscription_shared_ptr_const) {
+  rclcpp::init(0, nullptr);
```

Comment:
Calling `rclcpp::init` in each test is not a good idea. Currently it doesn't do anything but instead it should only be called once for all tests.